src/hotspot/os/aix/os_perf_aix.cpp

```diff
@@ -897,29 +575,64 @@ int CPUInformationInterface::cpu_information(CPUInformation& cpu_info) {
 }
 
 class NetworkPerformanceInterface::NetworkPerformance : public CHeapObj<mtInternal> {
-  friend class NetworkPerformanceInterface;
+  NONCOPYABLE(NetworkPerformance);
+
  private:
+  char* allocate_string(const char* str) const;
+
+  public:
   NetworkPerformance();
-  NONCOPYABLE(NetworkPerformance);
   bool initialize();
   ~NetworkPerformance();
   int network_utilization(NetworkInterface** network_interfaces) const;
 };
 
-NetworkPerformanceInterface::NetworkPerformance::NetworkPerformance() {
-
-}
+NetworkPerformanceInterface::NetworkPerformance::NetworkPerformance() {}
 
 bool NetworkPerformanceInterface::NetworkPerformance::initialize() {
-  return true;
+  return initialize_libperfstat();
 }
 
-NetworkPerformanceInterface::NetworkPerformance::~NetworkPerformance() {
-}
+NetworkPerformanceInterface::NetworkPerformance::~NetworkPerformance() {}
+
+int NetworkPerformanceInterface::NetworkPerformance::network_utilization(NetworkInterface** network_interfaces) const {
+  ResourceMark rm;
+  int n_records = 0;
+  perfstat_netinterface_t* net_stats;
+  perfstat_id_t name_holder;
+  int records_allocated = 0;
+
+  assert(network_interfaces != NULL, "network_interfaces is NULL");
+
+  *network_interfaces = NULL;
+  strncpy(name_holder.name , FIRST_NETINTERFACE, IDENTIFIER_LENGTH);
+
+  // calling perfstat_<subsystem>(NULL, NULL, _, 0) returns number of available records
+  if ((n_records = libperfstat::perfstat_netinterface(NULL, NULL, sizeof(perfstat_netinterface_t), 0)) <= 0) {
```

Comment:
is n=0 an error? Could there conceivably be zero interfaces?

Reply:
Good point. I agree it's not reasonable to return an error if there are simply no network interfaces.